mtp_noms_ops/apps/security/tests/test_forms.py

```diff
@@ -2791,3 +2791,98 @@ def test_reject_with_api_error(self):
                 form.errors,
                 {'__all__': ['There was an error with your request.']},
             )
+
+
+class AssignCheckToUserFormTestCase(SimpleTestCase):
+    """
+    Tests related to the AssignCheckToUserForm.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.request = mock.MagicMock(
+            user=mock.MagicMock(
+                pk=5,
+                token=generate_tokens(),
+            ),
+        )
+
+    def test_assign_or_unassign(self):
+        """
+        Test that the form makes the right API call to assign the check to a user list.
+        """
+        check_id = 1
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                rsps.PATCH,
+                api_url(f'/security/checks/{check_id}/'),
+                status=204,
+                json={
+                    'assigned_to': 5
+                }
+            )
+
+            form = AssignCheckToUserForm(
+                object_id=check_id,
+                request=self.request,
+                data={
+                    'assignment': 'assign',
+                },
+            )
+
+            self.assertTrue(form.is_valid())
+            self.assertEqual(form.assign_or_unassign(), True)
+
+    def test_unassign(self):
+        """
+        Test that the form makes the right API call to unassign the check from a user list.
+        """
+        check_id = 1
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                rsps.PATCH,
+                api_url(f'/security/checks/{check_id}/'),
+                status=204,
+                json={
+                    'assigned_to': None
+                }
+            )
+
+            form = AssignCheckToUserForm(
+                object_id=check_id,
+                request=self.request,
+                data={
+                    'assignment': 'unassign',
+                },
+            )
+
+            self.assertTrue(form.is_valid())
+            self.assertEqual(form.assign_or_unassign(), True)
+
+    def test_form_returns_error_if_check_already_assigned_to_other_user(self):
+        """
+        Test that the form returns an error if check is assigned to another user.
+        """
+        check_id = 1
+        with responses.RequestsMock() as rsps, silence_logger():
+            rsps.add(
+                rsps.PATCH,
+                api_url(f'/security/checks/{check_id}/'),
+                status=400
+            )
+
+            form = AssignCheckToUserForm(
+                object_id=check_id,
+                request=self.request,
+                data={
+                    'assignment': 'assign',
+                },
+            )
+
+            form.is_valid()
+
+            self.assertEqual(form.assign_or_unassign(), False)
+            self.assertDictEqual(
+                form.errors,
```

Comment:
while this test may pass, these errors wouldn't be displayed anywhere cos the view redirects regardless of whether the form is valid or not.